dictionary-builder/src/main.rs

```diff
@@ -57,285 +60,199 @@ fn main() {
+            match format {
+                "rs" => to_code_file(dst, entries, !ignore_retired),
+                "json" => to_json_file(dst, entries),
+                _ => unreachable!(),
+            }
+            .expect("Failed to write file");
+        }
+    } else if let Some(matches) = matches.subcommand_matches("keys") {
+        let out_file = matches.value_of("OUTPUT").unwrap();
+        let dst = Path::new(out_file);
+
+        let mut core = Core::new().unwrap();
+
+        if src.starts_with("http:") || src.starts_with("https:") {
+            let src = Uri::from_str(src).unwrap();
+            println!("Downloading DICOM dictionary ...");
+            let req = dic_from_site(src).and_then(|resp| {
+                resp.into_body().concat2().and_then(|body: Chunk| {
+                    let entries = parse_entries(&*body).unwrap();
+                    println!("Writing to file ...");
+                    to_keys_code_file(dst, entries, !ignore_retired).expect("failed to write file");
+                    Ok(())
+                })
+            });
+            core.run(req).unwrap();
+        } else {
+            // read from File
+            let file = File::open(src).unwrap();
+            let entries = parse_entries(BufReader::new(file)).unwrap();
+            println!("Writing to file ...");
+            to_keys_code_file(dst, entries, !ignore_retired).expect("failed to write file");
         }
-        .expect("Failed to write file");
     }
 }
 
-type XmlResult<T> = Result<T, XmlError>;
 type DynResult<T> = Result<T, Box<dyn std::error::Error>>;
 
-fn xml_from_site(url: Uri) -> ResponseFuture {
+fn dic_from_site(url: Uri) -> ResponseFuture {
     let client = Client::new();
     client.get(url)
 }
 
+fn parse_entries<R: BufRead>(source: R) -> DynResult<Vec<Entry>> {
+    let mut result = vec![];
+
+    for line in source.lines() {
+        let line = line?;
+        if line.starts_with('#') {
+            continue;
+        }
+
+        let parts: Vec<&str> = line.split('\t').collect();
+
+        let mut vr = parts[1].to_string();
+
+        // Some fields are dependent upon context.
+        // We may want to support this at some point, but for now,
+        // let's just choose some good defaults.
+        if vr == "up" {
+            vr = "UL".to_string();
+        }
+        if vr == "xs" {
+            vr = "US".to_string();
+        }
+        if vr == "ox" {
+            vr = "OB".to_string();
+        }
+        if vr == "lt" {
+            vr = "OW".to_string();
+        }
+        if vr == "na" {
+            // These are the "Item", "ItemDelimitationItem", "SequenceDelimitationItem", etc, values.
+            // Question, should we generate const values for these, and use them internally?
+            continue;
+        }
+
+        result.push(Entry {
+            tag: parts[0].to_string(),
+            name: parts[2].to_string(),
+            alias: parts[2].to_string(),
```

Comment:
`name` and `alias` have the same representation with this change. If recovering the other, more human readable names is not possible, it might be time to also rethink the `Entry` struct.